src/rt/sched/cown.h

```diff
@@ -137,6 +160,13 @@ namespace verona::rt
      **/
     std::atomic<size_t> weak_count{1};
 
+    /*
+     * Cown's read ref count.
```

Comment:
Could you make a class that encapsulates this logic?

In particular, there are just a few operations

* `fetch_add(2)` - add read
* try_write - true for no readers, false marks that a writer is waiting.
* release_read 

Then all this logic can be nicely encapsulated in one spot.  Currently it is spread out a bit more than might be easy to follow.